mzLib/MassSpectrometry/BaseClasses/IProduct.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Chemistry;
+
+namespace MassSpectrometry
+{
+    public interface IProduct : IHasMass, IEquatable<IProduct>
+    {
+        public double NeutralMass { get; }
+        public ProductType ProductType { get; }
+        public double NeutralLoss { get; }
+        public FragmentationTerminus Terminus { get; }
+        public int FragmentNumber { get; }
+        public int ResiduePosition { get; } // added for interface compatibility 
+        public int AminoAcidPosition => ResiduePosition;
+        public ProductType? SecondaryProductType { get; } //used for internal fragment ions
+        public int SecondaryFragmentNumber { get; } //used for internal fragment ions
```

Comment:
should this be nullable? if there is no secondary product type

Reply:
In the original implementation, this was not nullable, with the default set to zero. This behavior has been maintained as to not throw a bunch of null check warnings in MM when updated to this mzLib